src/components/screens/new-questionnaire-screen.tsx

```diff
@@ -7,48 +22,90 @@ import { KeyboardReturn } from "@mui/icons-material";
   const adminMode = UserRoleUtils.adminMode();
-  
+  const { questionnaireApi } = useLambdasApi();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
   const [questionnaireTitle, setQuestionnaireTitle] = useState("");
+  const [questionnaireDescription, setQuestionnaireDescription] = useState("");
+  const [options, setOptions] = useState<Question[]>([]);
+  const [passScoreValue, setPassScoreValue] = useState(0);
 
   const handleQuestionTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
     setQuestionnaireTitle(event.target.value);
   };
 
-  /**
-   * State and functions to handle the questions and options
-   */
-  const [questions, setQuestions] = useState<Question[]>([]);
+  const handleDescriptionChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuestionnaireDescription(event.target.value);
+  };
+
+  const handlePassScoreSliderChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setPassScoreValue(Number(event.target.value));
+  };
 
   /**
    * Functions to add new question to Questionnaire
-   * @param questionText 
-   * @param options 
+   * @param question
+   * @param options
    */
-  const handleAddQuestionSubmit = (
-    questionText: string,
-    options: QuestionOption[]
-  ) => {
-    setQuestions((prevQuestions: Question[]) => [...prevQuestions, { questionText, options }]);
+  const handleAddQuestionSubmit = (question: string, options: QuestionOption[]) => {
+    setOptions((prevQuestions: Question[]) => [...prevQuestions, { question, options }]);
   };
 
   const handleDeleteQuestion = (index: number) => {
-    const updatedQuestions = [...questions];
+    const updatedQuestions = [...options];
     updatedQuestions.splice(index, 1);
-    setQuestions(updatedQuestions);
+    setOptions(updatedQuestions);
+  };
+
+  /**
+   * Function to count all correct answers in the questionnaire, used for passScore determination
+   */
+  const countCorrectAnswers = () => {
+    return options.reduce((count, question) => {
+      return count + question.options.filter((option) => option.value === true).length;
+    }, 0);
   };
 
   /**
-   * This should save the question title and questions to DB.
-   * TODO:
+   * Function to save the new questionnaire
    */
-  const handleSaveSubmit = () => {};
+  const handleSaveSubmit = async () => {
+    try {
+      setLoading(true);
+      setError(null);
+
+      const createdQuestionnaire = await questionnaireApi.createQuestionnaire({
+        questionnaire: {
+          id: "",
+          title: questionnaireTitle,
+          description: questionnaireDescription,
+          options: options,
+          passScore: passScoreValue
+        }
+      });
+      return createdQuestionnaire;
+    } catch (error) {
+      console.error("Failed to save questionnaire", error);
+      setError("Failed to save questionnaire. Please try again.");
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
```

Comment:
Hopefully use of the error handler can handle all of this. Also the Error string should come from the localized strings files same as all other error strings